.cppcheck/includes.txt

```diff
@@ -0,0 +1,92 @@
+/concord-bft/
```

Comment:
whether you using cmake -DCMAKE_CXX_CPPCHECK or -DCMAKE_EXPORT_COMPILE_COMMANDS you don't want to maintain this file

Reply:
The exported file compile_commands.json is limited, it does not point to all includes, some will miss.
It doesn't recognize system and 3rd party includes and some includes will be missing, we get warning by cppcheck pre processor, resuting in a non optimized analaysis. So this include file is more like a hint.
For now, I didn't find any better solution. Omitting this file will result in less optimizaed results.
Also, this file is mandatory if I want to call to for checks like this:
```
make cppcheck CPPCHECK_REL_TARGET_PATH=make cppcheck CPPCHECK_REL_TARGET_PATH=bftengine/src/preprocessor/
```
to check only preprocessor, or
```
make cppcheck CPPCHECK_REL_TARGET_PATH=bftengine/src/preprocessor/PreProcessor.cpp
```
to check a single file
or
```
make cppcheck CPPCHECK_REL_TARGET_PATH=bftengine/src/preprocessor/PreProcessor*
```
to check all files which start with 'PreProcessor'.
In all of these examples I cant use compile comands and I must use the includes.

I agree that solution is not perfect, but cppcheck itself isn't perfect...
BTW, in clang-tidy i think we are not even importing compile comands, it uses a linux search ...

Reply:
@glevkovich Does not using this file affects the tool's results when run via cmake? If so, please document that the script produces a more verbose result. If not, please document that the script is equivalent to running via cmake, but doesn't require compilation.